### ISSUE TITLE
ShardingAlgorithmProvidedBeanRegistry  login has failed

### ISSUE THREAD

**Issue by poo0054** [apache/shardingsphere#18966]
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
5.1.2

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-JDBC

### Expected behavior
ShardingAlgorithmProvidedBeanRegistry  login was successful .ShardingSphereAlgorithm#init() Call successful

### Actual behavior
ShardingAlgorithmProvidedBeanRegistry     login has failed  Causes the abstractalgorithmprovidedbeanregistry\postprocessafterinitialization method call to fail

### Reason analyze (If you can)

Shardingalgorithmprovidedbeanregistry is not injected into the springboot post processor. The processor does not exist in the abstractbeanfactory#getbeanpostprocessors() method

springboot version 2.3.10

![image](https://user-images.githubusercontent.com/85894494/177976032-3810b470-d2a5-48fc-ba28-a2186c708d22.png)

**Comment by linghengqian** [comment]
- I realize you're using the SpringBoot OSS version that stopped maintenance. Can you change to SpringBoot OSS 2.6 or 2.7 to reproduce? In SpringBoot OSS 2.5, for significant changes to the data source, refer to its Release Note. 

- I can't reproduce this problem locally.

**Comment by zjcnb** [comment]
@poo0054 Hi, can you provide a demo project?

**Comment by poo0054** [comment]
Hello, thank you very much for your reply. I think I've found the reason   .I should have introduced sentinel.
I found the sentinelbeanpostprocessor in sentinel. It is not a postprocessor decorated with static.   So sentinelautoconfiguration will be loaded. When creating sentinelautoconfiguration bean, many processors will be used, including methodvalidationpostprocessor.   And I just customized a bean validator in baseconfiguration, and I injected a testservice in baseconfiguration. I injected data sources into testservice. As a result, the data source is loaded in advance. The whole spring lifecycle sequence is completely destroyed



The following is my simplified case in gitlab:

> https://gitlab.com/poo0054/test

Thank you very much for your help!
